### ISSUE TITLE
[Java] Incorrect packages in JCenter/Bintray leading to inconsistent behavior

### ISSUE THREAD

**Issue by tameraw** [Azure/azure-iot-sdk-java#17]
_From @dluc on November 30, 2016 0:20_

Some IDEs, e.g. Scala developers using SBT, are commonly setup to look for packages also in **JCenter/Bintray**, other than **Maven Central**. In this case some developers will experience odd SDK behavior, because the versions published are not consistent. 

For instance, it took several hours to debug an issue that was affecting only Scala (missing classes loaded by `SSlEngineFacadeFactory`), to finally discover that Scala was pulling **iothub-java-service-client 1.0.10** from Bintray and using **Proton-J 0.12.2**, while instead Java was pulling from Maven Central and using **Proton-J 0.15.0**.

Please either remove the wrong packages or make sure they are always in sync. For instance you could publish only snapshots to Bintray, with a "-DEV" suffix (e.g. `iothub-java-service-client-1.0.10-DEV`).

* https://oss.sonatype.org/content/repositories/releases/com/microsoft/azure/iothub-java-client/iothub-java-service-client/1.0.10/iothub-java-service-client-1.0.10.pom

```xml
    <dependency>
      <groupId>org.apache.qpid</groupId>
      <artifactId>proton-j</artifactId>
      <version>0.15.0</version>
      <type>jar</type>
    </dependency>
```

* http://jcenter.bintray.com/com/microsoft/azure/iothub-java-client/iothub-java-service-client/1.0.10/:iothub-java-service-client-1.0.10.pom

```xml
    <dependency>
      <groupId>com.microsoft.azure.iot</groupId>
      <artifactId>proton-j-azure-iot</artifactId>
      <version>0.12.2</version>
      <type>jar</type>
    </dependency>
```

_Copied from original issue: Azure/azure-iot-sdks#1002_

**Comment by tameraw** [comment]
@dluc - We'll look into removing the packages from JCenter & Bintray

**Comment by tameraw** [comment]
@dluc - Please double check with the latest release. Thanks.

**Comment by dluc** [comment]
The files for `iothub-java-service-client` `1.0.11` in Maven Central and Bintray are fine (same hash).

Just one note: the Maven Central archive is richer than the one in Bintray, e.g. the javadoc files are missing in Bintray.  

Unless you plan to deploy snapshots (development or daily builds) in there, I think it's safe to drop the Bintray repository.

Thanks

**Comment by tameraw** [comment]
@dluc - It is true that for the 1.0.10 version of the SDK the POM files in Bintray and Maven are out of sync. This was due to the one incident where we accidentally overridden an existing Maven version and Bintray did not sync to that "overridden" version.
I've made a request to own the Bintray package for our SDK and will hopefully be able to delete it or add a note for that 1.0.10 alerting users to the discrepancy.
